wvd-sh/WVD scaling script/basicScale-MSI.ps1

```diff
@@ -571,24 +585,32 @@ if ($hostpoolInfo.LoadBalancerType -eq "DepthFirst") {
       
                         $getHosts = Get-RdsSessionHost -TenantName $tenantname -HostPoolName $hostpoolname | Sort-Object Sessions | Sort-Object Status
                         foreach ($sessionhost in $getHosts) {
-                            #if (!($sessionHost.Sessions -eq $hostpoolMaxSessionLimit)) {
-                            #if ($sessionHost.Sessions -ge $SessionhostLimit) {
-          
                             if ($sessionHost.Status -eq "Available" -and $sessionHost.Sessions -eq 0) 
                             { break }
-                            if ($sessionHost.Status -eq "NoHeartbeat") {
+                            #Check the session host status and session host is healthy or not before starting the host
+                            if ($sessionHost.Status -eq "NoHeartbeat" -and $sessionhost.UpdateState -eq "Succeeded") {
                                 Write-Log 1 "Existing Sessionhost Sessions value reached near by hostpool maximumsession limit need to start the session host" "Info"
-                                $sessionhostname = $sessionHost.sessionhostname
+                                $sessionhostname = $sessionHost.sessionhostname | Out-String
+                                
+                                $VMName = $sessionHostname.Split(".")[0]
+                                $VmInfo = Get-AzureRmVM | Where-Object { $_.Name -eq $VMName }
+                                #Check the Session host is in Maintainance
+                                if($VmInfo.Tags.Keys -eq $maintainanceTagName){
+                                Write-Log 1 "Session Host is in Maintainace: $sessionhostname"
```

Comment:
Replace "Maintainace" with "maintenance"

Reply:
We have replaced in both scripts